src/main/webapp/results/results.js

```diff
@@ -0,0 +1,199 @@
+      return (
+        <Typography style={{ ...cellStyles }}>{result[attribute]}</Typography>
+      )
+  }
+}
+
+const Result = props => {
+  const { attributes, selected, id, onClick, Thumbnail, ...rest } = props
+  return (
+    <TableRow onClick={onClick} key={id} selected={selected}>
+      {attributes.map(attribute => (
+        <TableCell key={attribute}>
+          {getCellContent(attribute.toLowerCase(), { id, ...rest }, Thumbnail)}
+        </TableCell>
+      ))}
+    </TableRow>
+  )
+}
+
+const DDFThumbnail = props => {
+  const src = props.actions.find(
+    ({ id }) => id === 'catalog.data.metacard.thumbnail'
+  )
+  return <DefaultThumbnail src={src} {...props} />
+}
+
+const computeSelected = (selection, results, start, end, e) => {
+  const clicked = results[end].id
+  if (e.ctrlKey || e.metaKey) {
+    return selection.has(clicked)
+      ? selection.remove(clicked)
+      : selection.add(clicked)
+  }
+  if (e.shiftKey && start !== null) {
+    const slice =
+      start < end
+        ? results.slice(start, end + 1)
+        : results.slice(end, start + 1)
+    const group = Set(slice.map(({ id }) => id))
+    return group.union(selection)
+  }
+  return selection.has(clicked) ? Set() : Set([results[end].id])
+}
+
+const Results = props => {
+  const { results, attributes, onSelect, Thumbnail = DDFThumbnail } = props
+  const selection = Set(props.selection)
+  const [lastSelected, setLastSelected] = useState(null)
+  const allowTextSelect = !useKeyPressed('Shift')
+
+  return (
+    <div style={{ height: '100%' }}>
+      <Toolbar style={{ height: '10%' }}>
+        <Typography variant="h5" component="h1">
+          Search Results
+        </Typography>
+      </Toolbar>
+      <Paper style={{ height: '90%', overflow: 'auto' }}>
+        <Table>
+          <TableHead>
+            <TableRow>
+              {attributes.map(attribute => (
+                <TableCell key={attribute}>
+                  <Typography>{attribute}</Typography>
+                </TableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody style={{ userSelect: allowTextSelect ? 'auto' : 'none' }}>
+            {results.map((result, i) => (
+              <Result
+                {...result}
+                Thumbnail={Thumbnail}
+                attributes={attributes}
+                selected={selection.has(result.id)}
+                onClick={e => {
+                  e.stopPropagation()
+                  const selected = computeSelected(
+                    selection,
+                    results,
+                    lastSelected,
+                    i,
+                    e
+                  )
+                  onSelect(selected)
+                  setLastSelected(e.shiftKey ? lastSelected : i)
+                }}
+              />
+            ))}
+          </TableBody>
+        </Table>
+      </Paper>
+    </div>
+  )
+}
+
+const mapStateToProps = state => ({
+  selected: getSelected(state),
```

Comment:
selection: getSelected(state) ? 
It doesn't look like you're getting selected from props in your Results component

Reply:
Nice catch, this should be `selection`